# HausaSentiLex comparison with others Natural Language Processing (NLP) libraries

# Intro

This article will go over the Python implementation of HausaSentiLex  with TextBlob, VADER, and naija-twitter-sentiment-afriberta-large perfomance analysis analysis.

After reading the article, you will learn
* What is HausaSentiLex?
* What is TextBlob?
* What is VADER?
* Comparison between HausaSentiLex with above listed models.

Let's get started!

# Step 1: Install And Import Python Libraries

The first step is to install and import Python libraries.
We need to install the `vaderSentiment` package for VADER package for flair.

In [ ]:
# Install libraries
!pip install transformers

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.7/7.7 MB 28.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 302.0/302.0 kB 37.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 81.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 74.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 295.0/295.0 kB 27.5 MB/s eta 0:00:00


In [ ]:
# Install vaderSentiment package for VADER
!pip install vaderSentiment

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 126.0/126.0 kB 3.0 MB/s eta 0:00:00


After installing the packages, let's import the Python libraries. We need to import `pandas` and `numpy` for data processing.

For the sentiment analysis, we need to import `TextBlob`, `SentimentIntensityAnalyzer` from `vaderSentiment`.

To check the sentiment prediction accuracy, we need to import `accuracy_score` from `sklearn`.

Last but not least, we set the `pandas` dataframe column width to be 1000, which will allow us to see more content from the review.

In [ ]:
# Data processing
import pandas as pd
import numpy as np

# Import TextBlob
from textblob import TextBlob

# Import VADER sentiment
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

# Import accuracy_score to check performance
from sklearn.metrics import accuracy_score

# Set a wider colwith
pd.set_option('display.max_colwidth', 1000)

# Step 2: Download And Read Data

The second step is to download and read in the dataset to be use for comparison.

The Datasets used for testing is NaijaSenti, a Lacuna Funded Project for the development of sentiment corpus for four Nigerian languages: Igbo, Hausa, Yoruba and Pidgin has the anotated data from three 4 widely spoken languages in Nigeria.we downloaded hau_tweets.txt via  https://github.com/hausanlp/NaijaSenti


We are able to download these datasets (Find attached in our code directory). Upload them to file.

Those who are using Google Colab for this analysis need to mount Google Drive to read the dataset. You can ignore the code below if you are not using Google Colab.

In [ ]:
# Mount Google Drive
from google.colab import drive
drive.mount('/content/drive')

# Change directory
import os
os.chdir("drive/MyDrive/HausaSentiLex")

# Print out the current directory
!pwd

Mounted at /content/drive
/content/drive/MyDrive/HausaSentiLex


**Evaluation Dataset** : Now let's read the data into a `pandas` dataframe and see what the dataset looks like.

In [ ]:
# Read in data
dataset_1 = pd.read_csv('hausa_tweets.csv', sep=',', names=['tweets', 'polarity'])

# Take a look at the data
dataset_1.head()

tweets  \
0                                                                                                                                    ‍️‍️‍️‍️ an haramta shanta ne a lokacin covid ďin ne   
1                                                                                                                                ‍️ ku dai yan chana baza ku bar mutane su zauna lafia ba   
2                                                                                                                                             ^^happy bday sir allah ya kara tsawon kwana   
3  ✓establishment of frsc nde ndlea nalda ✓abujakano express way ✓creation of states ✓abuja as fct ✓establishment of copyright law etc general allah ya kara lafiya da nisan kwana aameen   
4                                                            a a ka buge late comer kenan lallai angaisheka first in the history late comer yazo ya kasa komai u deserve a medal for this   

   polarity  
0        -1  
1        -1  
2         1  
3         1  
4         1

`.info` helps us to get the information about the dataset.

From the output, we can see that this data set has two columns, 1000 records, and no missing data. The 'tweets' column is `object` type, and the 'label'column is `int64` type.

In [ ]:
# Get the dataset information
dataset_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3514 entries, 0 to 3513
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   tweets    3514 non-null   object
 1   polarity  3514 non-null   int64 
dtypes: int64(1), object(1)
memory usage: 55.0+ KB


Next, let's check the distribution of the label. There are 1755 positive and 1759 negative tweets in the dataset, so we have a balanced dataset. For a balanced dataset, we can use accuracy as the performance metric.

In [ ]:
# Check the label distribution
dataset_1['polarity'].value_counts()

-1    1759
 1    1755
Name: polarity, dtype: int64

# Step 3: What is HausaSentiLex

HausaSentiLex is a pretrained lexicon low resources language model. The model was trained on Hausa Language (Hausa is a Chadic language spoken by the Hausa people in the northern half of Nigeria, Niger, Ghana, Cameroon, Benin and Togo, and the southern half of Niger, Chad and Sudan, with significant minorities in Ivory Coast. It is the most widely spoken language in West Africa, and one of the most widely spoken languages in Africa as a whole). The model has been shown to obtain competitive downstream performances on text classification on trained language


In [ ]:
# Use a pipeline as a high-level helper
from transformers import pipeline

HausaSentiLex_classifier = pipeline("text-classification", model="mangaphd/HausaSentiLex")

Some layers from the model checkpoint at mangaphd/HausaBERTa were not used when initializing TFBertForSequenceClassification: ['dropout_113']
- This IS expected if you are initializing TFBertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing TFBertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
All the layers of TFBertForSequenceClassification were initialized from the model checkpoint at mangaphd/HausaBERTa.
If your task is similar to the task the model of the checkpoint was trained on, you can already use TFBertForSequenceClassification for predictions without further training.


In [ ]:
# function for interpretation with score
def HausaSentiLex_sentiment(text):
  rs = HausaSentiLex_classifier(text)
  pts = rs[0]
  if(pts["score"] < 0.6):
    return 0
  else:
    if(pts["label"] == "LABEL_0"):
      return -1
    else:
      return 1

In [ ]:
# function for Text Classification Interpretation
def HausaSentiLex_sentiment_analysis(text):
  rs = HausaSentiLex_classifier(text)
  if(rs == 0):
    return 'Neutral'
  else:
    if(rs == -1):
      return 'Negative'
    else:
      return 'Positive'

We start by testing one sentence in Hausa language **dadinsa yakeji** meaning **he like its** in English. This sentence is actually classified as **POSITIVE**.

In [ ]:
# Example text
text = 'dadinsa yakeji'

# sentiment
HausaSentiLex_sentiment_analysis(text)

'Positive'

# Step 4: What is TextBlob?

TextBlob is a Python library for Natural Language Processing (NLP). Sentiment analysis is one of many NLP tasks that TextBlob supports.

The sentiment property in TextBlob returns a polarity score and a subjectivity score for the input text.
* The polarity score ranges from -1 to 1, where -1 means extremely negative, and 1 means highly positive. A score near 0 means neutral sentiment.
* The subjectivity score ranges from 0 to 1, where 0 means extremely objective and 1 means highly subjective.


We tested similar sentence Hausa language **dadinsa yakeji** meaning **he like its** in English. This sentence is actually classified as **POSITIVE**.
**TextBlob** Classified it as **NEUTRAL** while it is actually **POSITIVE**

In [ ]:
# Example text
text = 'dadinsa yakeji'

# TextBlob sentiment
textBlob_sentiment = TextBlob(text).sentiment
textBlob_sentiment

Sentiment(polarity=0.0, subjectivity=0.0)

We can use `.polarity` to extract the sentiment score.

In [ ]:
# Extract sentiment score
textBlob_sentiment.polarity

0.0

# Step 5: What is VADER?

VADER (Valence Aware Dictionary and sEntiment Reasoner) is a Python library focusing on social media sentiments. It has a built-in algorithm to change sentiment intensity based on punctuations, slang, emojis, and acronyms.

The output of VADER includes four scores: compound score, negative score, neutral score, and positive score.

* The pos, neu, and neg represent the percentage of tokens that fall into each category, so they add up together to be 100%.
* The compound score is a single score to measure the sentiment of the text. Similar to TextBlob, it ranges from -1 (extremely negative) to 1 (extremely positive). The scores near 0 represent the neural sentiment score.
* The compound score is not a simple aggregation of the pos, neu, and neg scores. Instead, it incorporates rule-based enhancements such as punctuation amplifiers.

In [ ]:
# Example text
text = 'dadinsa yakeji'

# VADER Sentiment
vader = SentimentIntensityAnalyzer()
vader_sentiment = vader.polarity_scores(text)
vader_sentiment

{'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}

VADER gave the sample text 'dadinsa yakeji.' compound score of 0. There is no negative word in the sentence, so the neg score value is 0. There are 100% of neutral words and 0% of positive words in the sentence.

The output of VADER is saved as a dictionary. We can extract the compound sentiment score by the key 'compound'.

**VADER also failed.**

In [ ]:
# Extract sentiment score
vader_sentiment['compound']

0.0

# Step 6: naija-twitter-sentiment-afriberta-large?
naija-twitter-sentiment-afriberta-large is a multilingual twitter sentiment classification model for four (4) Nigerian languages (Hausa, Igbo, Nigerian Pidgin, and Yorùbá) based on a fine-tuned castorini/afriberta_large large model. The output of the sentiment score is either Positive, Neutral or Negative

In [ ]:
# Use a pipeline as a high-level helper
from transformers import pipeline

afrilargeClassifier = pipeline("text-classification", model="Davlan/naija-twitter-sentiment-afriberta-large")

Next, we use predict to predict the sentiment.

The sample text '**dadinsa yakeji**.' has the predicted sentiment of POSITIVE.

In [ ]:
text = 'dadinsa yakeji'
prediction = afrilargeClassifier(text)
prediction[0]['label']

'positive'

# Step 7: How to Use HausaSentiLex for Bulk Sentiment Analysis

From section 7 downwards, we will test the models performance in terms of large dataset.

In [ ]:
# We start by installing Colorama to print colored text while lopping the text
!pip install colorama

In [ ]:
import csv
from colorama import Fore

In [ ]:
# Text & Evaluation on Dataset
accuracy_sum=0
num_tweets=0
#true positives  (TP)
TP =0
#false positives (FP)
FP =0
#true negatives (TN)
TN=0
#false negatives (FN)
FN=0
with open('hausa_tweets.csv', 'r') as csvfile:
    reader = csv.reader(csvfile, delimiter=',')
    for row in reader:
      num_tweets +=1
      mytweet=row[0]
      mylabel=int(row[1])
      HausaSentiLex_prediction = HausaSentiLex_sentiment(mytweet)
      if(HausaSentiLex_prediction == mylabel):
        accuracy_sum += 1
        print(Fore.GREEN + f"Successful : {mytweet} , ACTUAL {mylabel} :: HausaSentiLex {HausaSentiLex_prediction}")
        if(mylabel == 1):
           TP = TP + 1
        elif (mylabel == -1):
           TN =TN +1
      else:
        print (Fore.RED + f"Unsuccessful : {mytweet} , ACTUAL {mylabel} :: HausaSentiLex {HausaSentiLex_prediction}")
        if(mylabel == 1 and HausaSentiLex_prediction == -1):
          FP = FP + 1
        elif (mylabel == -1 and HausaSentiLex_prediction == 1):
          FN =FN +1

In [ ]:
HausaSentiLex_accuracy = (accuracy_sum/num_tweets)*100
print(f"Acuracy : {HausaSentiLex_accuracy} ")
print(f"True Positive : {TP}")
print(f"True Negative : {TN}")
print(f"False Positive : {FP}")
print(f"False Negative : {FN}")

precision = TP / (TP + FP)
recall = TP / (TP + FN)
F1_score = 2 * (precision * recall) / (precision + recall)

print(f"Precision : {precision}")
print(f"Recall : {recall}")
print(f"F1 - Score : {F1_score}")


Acuracy : 67.52988047808765 
True Positive : 960
True Negative : 1413
False Positive : 696
False Negative : 285
Precision : 0.5797101449275363
Recall : 0.7710843373493976
F1 - Score : 0.6618407445708376


**HausaSentiLex** accurately detect 98% on Naija Senti Dataset

# Step 8: How To Use TextBlob For Sentiment Analysis on bulk tweets

In step 8, we will apply TextBlob to the Amazon review dataset and see how it performs.

We first get the sentiment polarity for each review and save the values into a column called 'scores_TextBlob'. Then check if the polarity score is positive. If the score is greater than or equal to zero, the predicted sentiment for the review is positive (labeled as 1). Otherwise, the predicted sentiment for the review is negative (labeled as 0).

In [ ]:
# Test on Dataset
accuracy_sum3=0
num_tweets3=0
#true positives  (TP)
TP =0
#false positives (FP)
FP =0
#true negatives (TN)
TN=0
#false negatives (FN)
FN=0
with open('hausa_tweets.csv', 'r') as csvfile:
    reader = csv.reader(csvfile, delimiter=',')
    for row in reader:
      num_tweets3 +=1
      mytweet=row[0]
      mylabel=int(row[1])
      TextBlob_prediction = int(TextBlob(mytweet).sentiment.polarity)
      if (TextBlob_prediction == mylabel):
        print(Fore.GREEN + f"Successful : {mytweet} , ACTUAL {mylabel} :: TextBlob {TextBlob_prediction}")
        accuracy_sum3 += 1
        if(mylabel == 1):
           TP = TP + 1
        elif (mylabel == -1):
           TN =TN +1
      else:
        print (Fore.RED + f"Unsuccessful : {mytweet} , ACTUAL {mylabel} :: TextBlob {TextBlob_prediction}")
        if(mylabel == 1 and TextBlob_prediction == -1):
          FP = FP + 1
        elif (mylabel == -1 and TextBlob_prediction == 1):
          FN =FN +1

Unsuccessful : ‍️‍️‍️‍️ an haramta shanta ne a lokacin covid ďin ne , ACTUAL -1 :: TextBlob 0
Unsuccessful : ‍️ ku dai yan chana baza ku bar mutane su zauna lafia ba , ACTUAL -1 :: TextBlob 0
Unsuccessful : ^^happy bday sir allah ya kara tsawon kwana , ACTUAL 1 :: TextBlob 0
Unsuccessful : ✓establishment of frsc nde ndlea nalda ✓abujakano express way ✓creation of states ✓abuja as fct ✓establishment of copyright law etc general allah ya kara lafiya da nisan kwana aameen , ACTUAL 1 :: TextBlob 0
Unsuccessful : a a ka buge late comer kenan lallai angaisheka first in the history late comer yazo ya kasa komai u deserve a medal for this , ACTUAL 1 :: TextBlob 0
Unsuccessful : a a mutuniyar fa yau ansa hijab gashi har kinyi kyau dole sema munje india gun , ACTUAL 1 :: TextBlob 0
Unsuccessful : a bincikesu dakyau wlh duk wanda aka kama da almundahna da dukiyar mu a kulle , ACTUAL -1 :: TextBlob 0
Unsuccessful : a dai yi mu gani idan tusaa zata hura wuta , ACTUAL -1 :: TextBlob 0
Unsuccessful 

In [ ]:
TextBlob_accuracy = (accuracy_sum3/num_tweets3)*100
print(f"Acuracy : {TextBlob_accuracy} ")
print(f"True Positive : {TP}")
print(f"True Negative : {TN}")
print(f"False Positive : {FP}")
print(f"False Negative : {FN}")

precision = TP / (TP + FP)
recall = TP / (TP + FN)
F1_score = 2 * (precision * recall) / (precision + recall)

print(f"Precision : {precision}")
print(f"Recall : {recall}")
print(f"F1 - Score : {F1_score}")

Acuracy : 0.512236767216847 
True Positive : 13
True Negative : 5
False Positive : 0
False Negative : 1
Precision : 1.0
Recall : 0.9285714285714286
F1 - Score : 0.962962962962963


After getting predictions from TextBlob, let's check the prediction accuracy.

Comparing the actual label with the TextBlob prediction on NaijaSenti Dataset, we get an accuracy score of 0.51, which means that TextBlob predicted the review sentiment 0.51% of the time.

# Step 9: How To Use VADER For Sentiment Analysis

In step 9, we will apply VADER to the evaluation dataset and see how it performs.

We first get the sentiment compound score for each review and save the values into a column called 'scores_VADER'. Then check if the compound score is positive. If the score is greater than or equal to zero, the predicted sentiment for the review is positive (labeled as 1). Otherwise, the predicted sentiment for the review is negative (labeled as 0).

In [ ]:
# Test on Dataset
accuracy_sum6=0
num_tweets6=0;
#true positives  (TP)
TP =0
#false positives (FP)
FP =0
#true negatives (TN)
TN=0
#false negatives (FN)
FN=0
with open('hausa_tweets.csv', 'r') as csvfile:
    reader = csv.reader(csvfile, delimiter=',')
    for row in reader:
      num_tweets6 +=1
      mytweet=row[0]
      mylabel=int(row[1])
      vader_s =0;
      vader = SentimentIntensityAnalyzer()
      vader_sentiment = vader.polarity_scores(mytweet)

      if (int(vader_sentiment['neg']) == 1):
          vader_s = -1
      elif (int(vader_sentiment['pos']) == 1):
        vader_s = 1
      else:
        vader_s = 0

      vader_prediction = vader_s

      if(vader_prediction == mylabel):
        print(Fore.GREEN + f"Successful : {mytweet} , ACTUAL {mylabel} :: Vader {vader_prediction}")
        accuracy_sum6 += 1
        if(mylabel == 1):
           TP = TP + 1
        elif (mylabel == -1):
           TN =TN +1
      else:
        print (Fore.RED + f"Unsuccessful : {mytweet} , ACTUAL {mylabel} :: Vader {vader_prediction}")
        if(mylabel == 1 and vader_prediction == -1):
          FP = FP + 1
        elif (mylabel == -1 and vader_prediction == 1):
          FN =FN +1

Unsuccessful : ‍️‍️‍️‍️ an haramta shanta ne a lokacin covid ďin ne , ACTUAL -1 :: Vader 0
Unsuccessful : ‍️ ku dai yan chana baza ku bar mutane su zauna lafia ba , ACTUAL -1 :: Vader 0
Unsuccessful : ^^happy bday sir allah ya kara tsawon kwana , ACTUAL 1 :: Vader 0
Unsuccessful : ✓establishment of frsc nde ndlea nalda ✓abujakano express way ✓creation of states ✓abuja as fct ✓establishment of copyright law etc general allah ya kara lafiya da nisan kwana aameen , ACTUAL 1 :: Vader 0
Unsuccessful : a a ka buge late comer kenan lallai angaisheka first in the history late comer yazo ya kasa komai u deserve a medal for this , ACTUAL 1 :: Vader 0
Unsuccessful : a a mutuniyar fa yau ansa hijab gashi har kinyi kyau dole sema munje india gun , ACTUAL 1 :: Vader 0
Unsuccessful : a bincikesu dakyau wlh duk wanda aka kama da almundahna da dukiyar mu a kulle , ACTUAL -1 :: Vader 0
Unsuccessful : a dai yi mu gani idan tusaa zata hura wuta , ACTUAL -1 :: Vader 0
Unsuccessful : a daidai wannan lokaci

Comparing the actual label with the VADER prediction, we get an accuracy score of 0.0, which means that VADER predicted the review sentiment 0% of the time.

In [ ]:
Vader_accuracy = (accuracy_sum6/num_tweets6)*100
print(f"Acuracy : {Vader_accuracy} ")
print(f"True Positive : {TP}")
print(f"True Negative : {TN}")
print(f"False Positive : {FP}")
print(f"False Negative : {FN}")

precision = TP / (TP + FP)
recall = TP / (TP + FN)
F1_score = 2 * (precision * recall) / (precision + recall)

print(f"Precision : {precision}")
print(f"Recall : {recall}")
print(f"F1 - Score : {F1_score}")

Acuracy : 0.0 
True Positive : 0
True Negative : 0
False Positive : 0
False Negative : 0


ZeroDivisionError: ignored

Comparing the actual label with the VADER prediction on Dataset, we get an accuracy score of 0.0.

# Step 10: How To Use naija-twitter-sentiment-afriberta-large For Sentiment Analysis


In step 10, we will apply the **naija-twitter-sentiment-afriberta-large** pre-trained sentiment model to predict the sentiment of the 3 Datasets.

Firstly, let's define a function that takes a review as input and the predicted label and confidence as outputs.


In [ ]:
# Test on Dataset 1
accuracy_sum12=0
num_tweets12=0
#true positives  (TP)
TP =0
#false positives (FP)
FP =0
#true negatives (TN)
TN=0
#false negatives (FN)
FN=0
with open('hausa_tweets.csv', 'r') as csvfile:
    reader = csv.reader(csvfile, delimiter=',')
    for row in reader:
      num_tweets12 +=1
      mytweet=row[0]
      mylabel=int(row[1])
      rs = afrilargeClassifier(mytweet)
      if(rs[0]['label'] == 'positive'):
        afriberta_prediction = 1
      elif(rs[0]['label'] == 'negative'):
        afriberta_prediction = -1
      else:
        afriberta_prediction = 0
      if (afriberta_prediction == mylabel):
        print(Fore.GREEN + f"Successful : {mytweet} , ACTUAL {mylabel} :: afriberta {afriberta_prediction}")
        accuracy_sum12 += 1
        if(mylabel == 1):
           TP = TP + 1
        elif(mylabel == -1):
           TN =TN +1
      else:
        print (Fore.RED + f"Unsuccessful : {mytweet} , ACTUAL {mylabel} :: afriberta {afriberta_prediction}")
        if(mylabel == 1 and vader_prediction == -1):
          FP = FP + 1
        elif(mylabel == -1 and vader_prediction == 1):
          FN =FN +1

Successful : ‍️‍️‍️‍️ an haramta shanta ne a lokacin covid ďin ne , ACTUAL -1 :: afriberta -1
Unsuccessful : ‍️ ku dai yan chana baza ku bar mutane su zauna lafia ba , ACTUAL -1 :: afriberta 0
Successful : ^^happy bday sir allah ya kara tsawon kwana , ACTUAL 1 :: afriberta 1
Successful : ✓establishment of frsc nde ndlea nalda ✓abujakano express way ✓creation of states ✓abuja as fct ✓establishment of copyright law etc general allah ya kara lafiya da nisan kwana aameen , ACTUAL 1 :: afriberta 1
Successful : a a ka buge late comer kenan lallai angaisheka first in the history late comer yazo ya kasa komai u deserve a medal for this , ACTUAL 1 :: afriberta 1
Successful : a a mutuniyar fa yau ansa hijab gashi har kinyi kyau dole sema munje india gun , ACTUAL 1 :: afriberta 1
Unsuccessful : a bincikesu dakyau wlh duk wanda aka kama da almundahna da dukiyar mu a kulle , ACTUAL -1 :: afriberta 1
Successful : a dai yi mu gani idan tusaa zata hura wuta , ACTUAL -1 :: afriberta -1
Successful : a 

In [ ]:
afriberta_accuracy = (accuracy_sum12/num_tweets12)*100
print(f"Acuracy : {afriberta_accuracy} ")
print(f"True Positive : {TP}")
print(f"True Negative : {TN}")
print(f"False Positive : {FP}")
print(f"False Negative : {FN}")

precision = TP / (TP + FP)
recall = TP / (TP + FN)
F1_score = 2 * (precision * recall) / (precision + recall)

print(f"Precision : {precision}")
print(f"Recall : {recall}")
print(f"F1 - Score : {F1_score}")

Acuracy : 83.52305065452475 
True Positive : 1553
True Negative : 1382
False Positive : 0
False Negative : 0
Precision : 1.0
Recall : 1.0
F1 - Score : 1.0


Comparing the actual label with the Afriberta prediction on Dataset 1(Dataset provided by the author of the model), we get an accuracy score of 83.52, which means that Flair predicted the review sentiment 83% of the time.

We can see that the summary statistics show that most flair scores are close to 1, which means that the flair sentiment model is overall very confident about the predictions.

# References

* [NaijaSenti: A Nigerian Twitter Sentiment Corpus for Multilingual Sentiment Analysis](https://https://arxiv.org/abs/2201.08277)
* [Sentiment Labelled Sentences Data Set on UCI Machine Learning Repository](https://archive.ics.uci.edu/ml/datasets/Sentiment+Labelled+Sentences)
* [TextBlob Documentation](https://textblob.readthedocs.io/en/dev/)
* [VADER Documentation](https://github.com/cjhutto/vaderSentiment)
* [Flair Documentation](https://github.com/flairNLP/flair)
* [Flair Tutorial](https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_1_BASICS.md)
* [Hutto, C.J. & Gilbert, E.E. (2014). VADER: A Parsimonious Rule-based Model for Sentiment Analysis of Social Media Text. Eighth International Conference on Weblogs and Social Media (ICWSM-14). Ann Arbor, MI, June 2014.](http://comp.social.gatech.edu/papers/icwsm14.vader.hutto.pdf)
*[Dataset 1](https://github.com/hausanlp/NaijaSenti)
*[Dataset 2](https://ml-olympiad-hausa-sentiment-analysis-20/overview)
*[Dataset 3](https://autonlp.ai/redirect/cc100-hausa)
